File: backend/app/schemas/llm.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field


InsightType = Literal[
    "executive_summary",
    "data_gap_explanation",
    "investor_memo",
    "sensitivity_explanation",
    "document_qa",
]


class LlmGenerateRequest(BaseModel):
    model_name: str | None = None


class LlmInsightRead(BaseModel):
    id: str
    plant_id: str | None
    scenario_id: str | None
    document_id: str | None
    insight_type: InsightType
    model_name: str
    prompt: str
    response_text: str | None
    status: str
    error_message: str | None
    provider_response_id: str | None
    usage: dict[str, Any] | None = Field(default=None)
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/types/llm.ts
export type InsightType =
  | "executive_summary"
  | "data_gap_explanation"
  | "investor_memo"
  | "sensitivity_explanation"
  | "document_qa"
  | "prefeed_committee_brief";

export type LlmInsight = {
  id: string;
  plant_id: string | null;
  scenario_id: string | null;
  document_id: string | null;
  insight_type: InsightType;
  model_name: string;
  prompt: string;
  response_text: string | null;
  status: string;
  error_message: string | null;
  provider_response_id: string | null;
  usage: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
};

export type LlmGeneratePayload = {
  model_name?: string | null;
};
